src/sentry/silo/client.py

```diff
@@ -248,18 +247,16 @@ def request(
         if prefix_hash is not None:
             hash = sha1(f"{prefix_hash}{self.region.name}{method}{path}".encode()).hexdigest()
 
-        try:
-            response = super().request(
-                method=method,
-                path=path,
-                headers=headers,
-                data=data,
-                params=params,
-                json=json,
-                raw_response=raw_response,
-            )
-        except Exception:
-            self.check_request_attempts(hash=hash, method=method, path=path)
-            raise
+        self.check_request_attempts(hash=hash, method=method, path=path)
+        response = super().request(
+            method=method,
+            path=path,
+            headers=headers,
+            data=data,
+            params=params,
+            json=json,
+            raw_response=raw_response,
+        )
+
         self.cleanup_request_attempts(hash=hash)
```

Comment:
I'm still debating whether we want to leave this in or remove it. My thought here is that any request that succeeds 10 or more times within the request count cache TTL should also likely be cleaned up as it represents a stuck request.